src/transport/AdminPairingTable.cpp

```diff
@@ -27,41 +27,68 @@
 namespace chip {
 namespace Transport {
 
-CHIP_ERROR AdminPairingInfo::StoreIntoKVS(PersistentStorageDelegate & kvs)
+PersistentStorageDelegate * gStorage = nullptr;
+AdminPairingTableDelegate * gDelegate = nullptr;
+
+CHIP_ERROR AdminPairingInfo::StoreIntoKVS()
 {
+    if (gStorage == nullptr)
+    {
+        ChipLogError(Discovery, "Server storage delegate is nill, cannot store admin in KVS");
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
     char key[KeySize()];
     ReturnErrorOnFailure(GenerateKey(mAdmin, key, sizeof(key)));
 
     StorableAdminPairingInfo info;
     info.mNodeId = Encoding::LittleEndian::HostSwap64(mNodeId);
     info.mAdmin  = Encoding::LittleEndian::HostSwap16(mAdmin);
+    info.mFabricId = Encoding::LittleEndian::HostSwap64(mFabricId);
+    info.mVendorId = Encoding::LittleEndian::HostSwap16(mVendorId);
 
-    return kvs.SyncSetKeyValue(key, &info, sizeof(info));
+    gDelegate->OnAdminPersistedToStorage(mAdmin, mFabricId, mNodeId);
+    return gStorage->SyncSetKeyValue(key, &info, sizeof(info));
```

Comment:
This looks like it can end up with things out of sync, where we update the attr value but the storage is not updated...  Would it make sense to only call the delegate if the SyncSetKeyValue call succeeds?

Reply:
yes can fix.